src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseAdminController.java

```diff
@@ -143,17 +168,97 @@ public void fullDeleteStudy(Long[] ids) {
                     Pageable pageable,
             UserSearch userSearch)
             throws UserNotFoundException {
-
-        final Page<IUser> users = userService.getPage(pageable, userSearch);
+      
+        final Page<U> users = userService.getPage(pageable, userSearch);
         return users.map(user -> conversionService.convert(user, UserWithTenantsDTO.class));
     }
 
+    @ApiOperation("Register new users")
+    @RequestMapping(value = "/api/v1/admin/users/group", method = RequestMethod.POST)
+    public JsonResult<List<AdminUserDTO>> register(
+            @RequestBody @Valid BulkUsersRegistrationDTO bulkUsersDto
+    ) {
+
+        return saveUsers(bulkUsersDto);
+    }
+
+    private JsonResult<List<AdminUserDTO>> saveUsers(final BulkUsersRegistrationDTO bulkUsersDto) {
+
+        final JsonResult<List<AdminUserDTO>> result = new JsonResult<>(NO_ERROR);
+        final Map<String, Object> errorMessages = new HashMap<>();
+        if (CollectionUtils.isEmpty(bulkUsersDto.getTenantIds())) {
+            result.setErrorCode(VALIDATION_ERROR.getCode());
+            errorMessages.put(TENANTS_KEY_MESSAGE_ERROR, MUST_NOT_EMPTY_MESSAGE_ERROR);
+            result.setValidatorErrors(errorMessages);
+            return result;
+        }
+
+        final Set<Tenant> tenants = new HashSet<>(tenantService.findByIdsIn(bulkUsersDto.getTenantIds()));
+        final boolean emailConfirmationRequired = bulkUsersDto.getEmailConfirmationRequired();
+        List<CommonUserRegistrationDTO> userDtos = bulkUsersDto.getUsers();
+        List<AdminUserDTO> createdUsers = IntStream.range(0, userDtos.size())
+                .mapToObj(index -> {
+                    final CommonUserRegistrationDTO userDto = userDtos.get(index);
+                    U user = convert(userDto);
+                    updateFields(user, tenants, emailConfirmationRequired, bulkUsersDto.getPassword());
+                    U createdUser = createUserAndSetValidationError(user, index, result, errorMessages);
+                    sendRegistrationEmail(emailConfirmationRequired, userDto, createdUser);
+                    return createdUser;
+                })
+                .map(createdUser -> conversionService.convert(createdUser, AdminUserDTO.class))
+                .collect(Collectors.toList());
+
+        result.setResult(createdUsers);
+        if (result.getErrorCode().equals(VALIDATION_ERROR.getCode())) {
```

Comment:
potential PNE, it would be better to use VALIDATION_ERROR.getCode().equals(...), or Objects.equals